File: tests/fixtures/polyglot/sample.java
// Polyglot fixture — null-return-on-some-path.
// lookup() returns null when id is negative (line 8). Caller at line 15 derefs.
// Parses clean; spotbugs NP_NULL_ON_SOME_PATH would flag line 15.

public class Sample {
    static String lookup(int id) {
        if (id < 0) {
            return null;
        }
        return "user-" + id;
    }

    public static void main(String[] args) {
        String name = lookup(-1);
        System.out.println(name.length());
    }
}

File: tests/fixtures/polyglot/sample.rb
# Polyglot fixture — to_i on a value that could be nil.
# ENV['MISSING'] returns nil when unset; .to_i on nil raises NoMethodError.
# Parses clean; rubocop Lint/NilChecks / Lint/SafeNavigationChain would flag line 6.

def parse_port
  ENV['MISSING'].to_i
end

def main
  puts parse_port
end

main
